services/api/mw2pdf/utils/validation.js

```diff
@@ -0,0 +1,31 @@
+import Joi from 'joi'
+
+export function assertValidPassthroughParameters(passthroughParameters) {
+  Joi.assert(
+    passthroughParameters,
+    Joi.string()
+      .custom(
+        (value, helpers) => {
+          JSON.parse(value)
+          return value
+        },
+        'is a valid JSON string',
+      ),
+  )
+  Joi.assert(
+    JSON.parse(passthroughParameters),
+    Joi.object()
+      .pattern(
+        /\w+/,
+        Joi.any().custom(
+          (value, helpers) => {
+            if (typeof value === 'object' && value !== null) {
+              return helpers.error('any.invalid');
```

Comment:
And if the key's value is either not an object or is null, then we add an error for this validation?

Reply:
Decided to use the explicit `throw` variety of validation instead since I think it's a bit more obvious what's going on.